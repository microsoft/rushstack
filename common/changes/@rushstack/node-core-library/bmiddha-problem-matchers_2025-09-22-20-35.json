{
  "changes": [
    {
      "packageName": "@rushstack/node-core-library",
      "comment": "Add `FileError.getProblemMatcher()` which returns the problem matcher compatible with `IOperationExecutionResult.problemCollector` as well as VS Code, GitHub Actions",
      "type": "minor"
    }
  ],
  "packageName": "@rushstack/node-core-library"
}
