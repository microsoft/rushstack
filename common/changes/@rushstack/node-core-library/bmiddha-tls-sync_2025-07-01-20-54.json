{
  "changes": [
    {
      "packageName": "@rushstack/node-core-library",
      "comment": "Add a `Async.runWithTimeoutAsync` function that executes an async function, resolving if the specified timeout elapses first.",
      "type": "minor"
    }
  ],
  "packageName": "@rushstack/node-core-library"
}