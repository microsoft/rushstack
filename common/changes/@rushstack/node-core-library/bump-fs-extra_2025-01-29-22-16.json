{
  "changes": [
    {
      "packageName": "@rushstack/node-core-library",
      "comment": "Update fs-extra to 11.3.0.",
      "type": "minor"
    }
  ],
  "packageName": "@rushstack/node-core-library"
}