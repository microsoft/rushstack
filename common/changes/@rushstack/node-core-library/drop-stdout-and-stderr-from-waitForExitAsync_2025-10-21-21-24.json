{
  "changes": [
    {
      "packageName": "@rushstack/node-core-library",
      "comment": "Update the return type of `Executable.waitForExitAsync` to omit `stdout` and `stderr` if an `encoding` parameter isn't passed to the options object.",
      "type": "patch"
    }
  ],
  "packageName": "@rushstack/node-core-library"
}