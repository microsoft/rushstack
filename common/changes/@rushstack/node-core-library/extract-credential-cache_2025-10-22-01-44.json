{
  "changes": [
    {
      "packageName": "@rushstack/node-core-library",
      "comment": "Add \"Objects.areDeepEqual\" and \"User.getHomeFolder\" APIs.",
      "type": "minor"
    }
  ],
  "packageName": "@rushstack/node-core-library"
}