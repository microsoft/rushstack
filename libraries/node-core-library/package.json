{
  "name": "@rushstack/node-core-library",
  "version": "5.10.1",
  "description": "Core libraries that every NodeJS toolchain project should use",
  "main": "lib/index.js",
  "typings": "dist/node-core-library.d.ts",
  "license": "MIT",
  "repository": {
    "url": "https://github.com/microsoft/rushstack.git",
    "type": "git",
    "directory": "libraries/node-core-library"
  },
  "scripts": {
    "build": "heft build --clean",
    "_phase:build": "heft run --only build -- --clean",
    "_phase:test": "heft run --only test -- --clean"
  },
  "dependencies": {
    "fs-extra": "~7.0.1",
    "import-lazy": "~4.0.0",
    "jju": "~1.4.0",
    "resolve": "~1.22.1",
    "semver": "~7.5.4",
    "ajv": "~8.13.0",
    "ajv-draft-04": "~1.0.0",
    "ajv-formats": "~3.0.1"
  },
  "devDependencies": {
    "local-eslint-config": "workspace:*",
    "@rushstack/heft": "0.68.12",
    "@rushstack/heft-node-rig": "2.6.48",
    "@types/fs-extra": "7.0.0",
    "@types/heft-jest": "1.0.1",
    "@types/jju": "1.4.1",
    "@types/node": "18.17.15",
    "@types/resolve": "1.20.2",
    "@types/semver": "7.5.0"
  },
  "peerDependencies": {
    "@types/node": "*"
  },
  "peerDependenciesMeta": {
    "@types/node": {
      "optional": true
    }
  }
}
