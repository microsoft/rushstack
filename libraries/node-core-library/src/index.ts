// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

/**
 * Core libraries that every NodeJS toolchain project should use.
 *
 * @packageDocumentation
 */

export { AlreadyReportedError } from './AlreadyReportedError';
export {
  Async,
  AsyncQueue,
  type IAsyncParallelismOptions,
  type IRunWithRetriesOptions,
  type IWeighted
} from './Async';
export type { Brand } from './PrimitiveTypes';
export { FileConstants, FolderConstants } from './Constants';
export { Enum } from './Enum';
export { EnvironmentMap, type IEnvironmentEntry } from './EnvironmentMap';
export {
  type ExecutableStdioStreamMapping,
  type ExecutableStdioMapping,
  type IExecutableResolveOptions,
  type IExecutableSpawnSyncOptions,
  type IExecutableSpawnOptions,
  type IWaitForExitOptions,
  type IWaitForExitWithBufferOptions,
  type IWaitForExitWithStringOptions,
  type IWaitForExitResult,
  type IProcessInfo,
  Executable
} from './Executable';
export { type IFileErrorOptions, type IFileErrorFormattingOptions, FileError } from './FileError';
export type {
  INodePackageJson,
  IPackageJson,
  IPackageJsonDependencyTable,
  IPackageJsonScriptTable,
  IPackageJsonRepository,
  IPeerDependenciesMetaTable,
  IDependenciesMetaTable,
  IPackageJsonExports
} from './IPackageJson';
export {
  Import,
  type IImportResolveOptions,
  type IImportResolveAsyncOptions,
  type IImportResolveModuleOptions,
  type IImportResolveModuleAsyncOptions,
  type IImportResolvePackageOptions,
  type IImportResolvePackageAsyncOptions
} from './Import';
export { InternalError } from './InternalError';
export {
  type JsonObject,
  type JsonNull,
  JsonSyntax,
  type IJsonFileParseOptions,
  type IJsonFileLoadAndValidateOptions,
  type IJsonFileStringifyOptions,
  type IJsonFileSaveOptions,
  JsonFile
} from './JsonFile';
export {
  type IJsonSchemaErrorInfo,
  type IJsonSchemaFromFileOptions,
  type IJsonSchemaFromObjectOptions,
  type IJsonSchemaLoadOptions,
  type IJsonSchemaValidateOptions,
  type IJsonSchemaValidateObjectWithOptions,
  JsonSchema,
  type JsonSchemaVersion
} from './JsonSchema';
export { LockFile } from './LockFile';
export { MapExtensions } from './MapExtensions';
export { MinimumHeap } from './MinimumHeap';
export { PosixModeBits } from './PosixModeBits';
export { ProtectableMap, type IProtectableMapParameters } from './ProtectableMap';
export { type IPackageJsonLookupParameters, PackageJsonLookup } from './PackageJsonLookup';
export {
  PackageName,
  PackageNameParser,
  type IPackageNameParserOptions,
  type IParsedPackageName,
  type IParsedPackageNameOrError
} from './PackageName';
export {
  Path,
  type FileLocationStyle,
  type IPathFormatFileLocationOptions,
  type IPathFormatConciselyOptions
} from './Path';
export { Encoding, Text, NewlineKind, type IReadLinesFromIterableOptions } from './Text';
export { Sort } from './Sort';
export {
  AlreadyExistsBehavior,
  FileSystem,
  type FileSystemCopyFilesAsyncFilter,
  type FileSystemCopyFilesFilter,
  type FolderItem,
  type FileSystemStats,
  type IFileSystemCopyFileBaseOptions,
  type IFileSystemCopyFileOptions,
  type IFileSystemCopyFilesAsyncOptions,
  type IFileSystemCopyFilesOptions,
  type IFileSystemCreateLinkOptions,
  type IFileSystemDeleteFileOptions,
  type IFileSystemMoveOptions,
  type IFileSystemReadFileOptions,
  type IFileSystemReadFolderOptions,
  type IFileSystemUpdateTimeParameters,
  type IFileSystemWriteBinaryFileOptions,
  type IFileSystemWriteFileOptions
} from './FileSystem';
export { FileWriter, type IFileWriterFlags } from './FileWriter';
export { LegacyAdapters, type LegacyCallback } from './LegacyAdapters';
export { StringBuilder, type IStringBuilder } from './StringBuilder';
export { type ISubprocessOptions, SubprocessTerminator } from './SubprocessTerminator';
export { TypeUuid } from './TypeUuid';
