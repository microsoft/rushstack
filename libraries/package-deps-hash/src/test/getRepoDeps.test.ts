// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import * as path from 'path';
import { execSync } from 'child_process';

import {
  getDetailedRepoStateAsync,
  type IDetailedRepoState,
  parseGitLsTree,
  getRepoRoot,
  parseGitHashObject
} from '../getRepoState';

import { FileSystem } from '@rushstack/node-core-library';

const SOURCE_PATH: string = path.join(__dirname).replace(path.join('lib', 'test'), path.join('src', 'test'));

const TEST_PREFIX: string = `libraries/package-deps-hash/src/test/`;
const TEST_PROJECT_PATH: string = path.join(SOURCE_PATH, 'testProject');

const FILTERS: string[] = [`testProject/`, `nestedTestProject/`];

function checkSnapshot(results: IDetailedRepoState): void {
  const relevantResults: Record<string, string> = {};
  for (const [key, hash] of results.files) {
    if (key.startsWith(TEST_PREFIX)) {
      const partialKey: string = key.slice(TEST_PREFIX.length);
      relevantResults[partialKey] = hash;
    }
  }

  expect({
    hasSubmodules: results.hasSubmodules,
    hasUncommittedChanges: results.hasUncommittedChanges,
    files: relevantResults
  }).toMatchSnapshot();
}

describe(getRepoRoot.name, () => {
  it(`returns the correct directory`, () => {
    const root: string = getRepoRoot(__dirname);
    const expectedRoot: string = path.resolve(__dirname, '../../../..').replace(/\\/g, '/');
    expect(root).toEqual(expectedRoot);
  });
});

describe(parseGitLsTree.name, () => {
  it('can handle a blob', () => {
    const filename: string = 'src/typings/tsd.d.ts';
    const hash: string = '3451bccdc831cb43d7a70ed8e628dcf9c7f888c8';

    const output: string = `100644 blob ${hash}\t${filename}\x00`;
    const { files } = parseGitLsTree(output);

    expect(files.size).toEqual(1); // Expect there to be exactly 1 change
    expect(files.get(filename)).toEqual(hash); // Expect the hash to be ${hash}
  });

  it('can handle a submodule', () => {
    const filename: string = 'rushstack';
    const hash: string = 'c5880bf5b0c6c1f2e2c43c95beeb8f0a808e8bac';

    const output: string = `160000 commit ${hash}\t${filename}\x00`;
    const { submodules } = parseGitLsTree(output);

    expect(submodules.size).toEqual(1); // Expect there to be exactly 1 submodule change
    expect(submodules.get(filename)).toEqual(hash); // Expect the hash to be ${hash}
  });

  it('can handle multiple lines', () => {
    const filename1: string = 'src/typings/tsd.d.ts';
    const hash1: string = '3451bccdc831cb43d7a70ed8e628dcf9c7f888c8';

    const filename2: string = 'src/foo bar/tsd.d.ts';
    const hash2: string = '0123456789abcdef1234567890abcdef01234567';

    const filename3: string = 'submodule/src/index.ts';
    const hash3: string = 'fedcba9876543210fedcba9876543210fedcba98';

    const output: string = `100644 blob ${hash1}\t${filename1}\x00100666 blob ${hash2}\t${filename2}\x00106666 commit ${hash3}\t${filename3}\0`;
    const { files, submodules } = parseGitLsTree(output);

    expect(files.size).toEqual(2); // Expect there to be exactly 2 changes
    expect(files.get(filename1)).toEqual(hash1); // Expect the hash to be ${hash1}
    expect(files.get(filename2)).toEqual(hash2); // Expect the hash to be ${hash2}

    expect(submodules.size).toEqual(1); // Expect there to be exactly 1 submodule changes
    expect(submodules.get(filename3)).toEqual(hash3); // Expect the hash to be ${hash3}
  });
});

describe(parseGitHashObject.name, () => {
  it('can handle requesting zero entries', () => {
    const results: Map<string, string> = new Map(parseGitHashObject('', []));
    expect(results.size).toEqual(0);
  });

  it('can parse multiple entries', () => {
    const results: Map<string, string> = new Map(parseGitHashObject('11\n22\n33', ['a', 'b', 'c']));
    expect(results).toMatchSnapshot();
  });

  it('can parse multiple entries with trailing whitespace', () => {
    const results: Map<string, string> = new Map(parseGitHashObject('11\n22\n33\n\n', ['a', 'b', 'c']));
    expect(results).toMatchSnapshot();
  });

  it('throws if too few hashes are provided', () => {
    expect(() => {
      new Map(parseGitHashObject('11\n22', ['a', 'b', 'c']));
    }).toThrowErrorMatchingSnapshot();
  });

  it('throws if too many hashes are provided', () => {
    expect(() => {
      new Map(parseGitHashObject('11\n22\n33', ['a', 'b']));
    }).toThrowErrorMatchingSnapshot();
  });
});

describe(getDetailedRepoStateAsync.name, () => {
  it('can parse committed files', async () => {
    const results: IDetailedRepoState = await getDetailedRepoStateAsync(
      SOURCE_PATH,
      undefined,
      undefined,
      FILTERS
    );
    checkSnapshot(results);
  });

  it('can handle adding one file', async () => {
    const tempFilePath: string = path.join(TEST_PROJECT_PATH, 'a.txt');

    FileSystem.writeFile(tempFilePath, 'a');

    try {
      const results: IDetailedRepoState = await getDetailedRepoStateAsync(
        SOURCE_PATH,
        undefined,
        undefined,
        FILTERS
      );
      checkSnapshot(results);
    } finally {
      FileSystem.deleteFile(tempFilePath);
    }
  });

  it('can handle adding two files', async () => {
    const tempFilePath1: string = path.join(TEST_PROJECT_PATH, 'a.txt');
    const tempFilePath2: string = path.join(TEST_PROJECT_PATH, 'b.txt');

    FileSystem.writeFile(tempFilePath1, 'a');
    FileSystem.writeFile(tempFilePath2, 'a');

    try {
      const results: IDetailedRepoState = await getDetailedRepoStateAsync(
        SOURCE_PATH,
        undefined,
        undefined,
        FILTERS
      );
      checkSnapshot(results);
    } finally {
      FileSystem.deleteFile(tempFilePath1);
      FileSystem.deleteFile(tempFilePath2);
    }
  });

  it('can handle removing one file', async () => {
    const testFilePath: string = path.join(TEST_PROJECT_PATH, 'file1.txt');

    FileSystem.deleteFile(testFilePath);

    try {
      const results: IDetailedRepoState = await getDetailedRepoStateAsync(
        SOURCE_PATH,
        undefined,
        undefined,
        FILTERS
      );
      checkSnapshot(results);
    } finally {
      execSync(`git checkout --force HEAD -- ${TEST_PREFIX}testProject/file1.txt`, {
        stdio: 'ignore',
        cwd: getRepoRoot(SOURCE_PATH)
      });
    }
  });

  it('can handle changing one file', async () => {
    const testFilePath: string = path.join(TEST_PROJECT_PATH, 'file1.txt');

    FileSystem.writeFile(testFilePath, 'abc');

    try {
      const results: IDetailedRepoState = await getDetailedRepoStateAsync(
        SOURCE_PATH,
        undefined,
        undefined,
        FILTERS
      );
      checkSnapshot(results);
    } finally {
      execSync(`git checkout --force HEAD -- ${TEST_PREFIX}testProject/file1.txt`, {
        stdio: 'ignore',
        cwd: getRepoRoot(SOURCE_PATH)
      });
    }
  });

  it('can handle uncommitted filenames with spaces and non-ASCII characters', async () => {
    const tempFilePath1: string = path.join(TEST_PROJECT_PATH, 'a file.txt');
    const tempFilePath2: string = path.join(TEST_PROJECT_PATH, 'a  file name.txt');
    const tempFilePath3: string = path.join(TEST_PROJECT_PATH, 'newFile批把.txt');

    FileSystem.writeFile(tempFilePath1, 'a');
    FileSystem.writeFile(tempFilePath2, 'a');
    FileSystem.writeFile(tempFilePath3, 'a');

    try {
      const results: IDetailedRepoState = await getDetailedRepoStateAsync(
        SOURCE_PATH,
        undefined,
        undefined,
        FILTERS
      );
      checkSnapshot(results);
    } finally {
      FileSystem.deleteFile(tempFilePath1);
      FileSystem.deleteFile(tempFilePath2);
      FileSystem.deleteFile(tempFilePath3);
    }
  });

  it('handles requests for additional files', async () => {
    const tempFilePath1: string = path.join(TEST_PROJECT_PATH, 'log.log');

    FileSystem.writeFile(tempFilePath1, 'a');

    try {
      const results: IDetailedRepoState = await getDetailedRepoStateAsync(
        SOURCE_PATH,
        [`${TEST_PREFIX}testProject/log.log`],
        undefined,
        FILTERS
      );
      checkSnapshot(results);
    } finally {
      FileSystem.deleteFile(tempFilePath1);
    }
  });
});
