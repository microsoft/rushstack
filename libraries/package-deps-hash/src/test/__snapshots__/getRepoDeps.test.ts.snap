// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRepoStateAsync can handle adding one file 1`] = `
Object {
  "nestedTestProject/package.json": "18a1e415e56220fa5122428a4ef8eb8874756576",
  "nestedTestProject/src/file 1.txt": "c7b2f707ac99ca522f965210a7b6b0b109863f34",
  "testProject/a.txt": "2e65efe2a145dda7ee51d1741299f848e5bf752e",
  "testProject/file  2.txt": "a385f754ec4fede884a4864d090064d9aeef8ccb",
  "testProject/file1.txt": "c7b2f707ac99ca522f965210a7b6b0b109863f34",
  "testProject/file蝴蝶.txt": "ae814af81e16cb2ae8c57503c77e2cab6b5462ba",
  "testProject/package.json": "18a1e415e56220fa5122428a4ef8eb8874756576",
}
`;

exports[`getRepoStateAsync can handle adding two files 1`] = `
Object {
  "nestedTestProject/package.json": "18a1e415e56220fa5122428a4ef8eb8874756576",
  "nestedTestProject/src/file 1.txt": "c7b2f707ac99ca522f965210a7b6b0b109863f34",
  "testProject/a.txt": "2e65efe2a145dda7ee51d1741299f848e5bf752e",
  "testProject/b.txt": "2e65efe2a145dda7ee51d1741299f848e5bf752e",
  "testProject/file  2.txt": "a385f754ec4fede884a4864d090064d9aeef8ccb",
  "testProject/file1.txt": "c7b2f707ac99ca522f965210a7b6b0b109863f34",
  "testProject/file蝴蝶.txt": "ae814af81e16cb2ae8c57503c77e2cab6b5462ba",
  "testProject/package.json": "18a1e415e56220fa5122428a4ef8eb8874756576",
}
`;

exports[`getRepoStateAsync can handle changing one file 1`] = `
Object {
  "nestedTestProject/package.json": "18a1e415e56220fa5122428a4ef8eb8874756576",
  "nestedTestProject/src/file 1.txt": "c7b2f707ac99ca522f965210a7b6b0b109863f34",
  "testProject/file  2.txt": "a385f754ec4fede884a4864d090064d9aeef8ccb",
  "testProject/file1.txt": "f2ba8f84ab5c1bce84a7b441cb1959cfc7093b7f",
  "testProject/file蝴蝶.txt": "ae814af81e16cb2ae8c57503c77e2cab6b5462ba",
  "testProject/package.json": "18a1e415e56220fa5122428a4ef8eb8874756576",
}
`;

exports[`getRepoStateAsync can handle removing one file 1`] = `
Object {
  "nestedTestProject/package.json": "18a1e415e56220fa5122428a4ef8eb8874756576",
  "nestedTestProject/src/file 1.txt": "c7b2f707ac99ca522f965210a7b6b0b109863f34",
  "testProject/file  2.txt": "a385f754ec4fede884a4864d090064d9aeef8ccb",
  "testProject/file蝴蝶.txt": "ae814af81e16cb2ae8c57503c77e2cab6b5462ba",
  "testProject/package.json": "18a1e415e56220fa5122428a4ef8eb8874756576",
}
`;

exports[`getRepoStateAsync can handle uncommitted filenames with spaces and non-ASCII characters 1`] = `
Object {
  "nestedTestProject/package.json": "18a1e415e56220fa5122428a4ef8eb8874756576",
  "nestedTestProject/src/file 1.txt": "c7b2f707ac99ca522f965210a7b6b0b109863f34",
  "testProject/a  file name.txt": "2e65efe2a145dda7ee51d1741299f848e5bf752e",
  "testProject/a file.txt": "2e65efe2a145dda7ee51d1741299f848e5bf752e",
  "testProject/file  2.txt": "a385f754ec4fede884a4864d090064d9aeef8ccb",
  "testProject/file1.txt": "c7b2f707ac99ca522f965210a7b6b0b109863f34",
  "testProject/file蝴蝶.txt": "ae814af81e16cb2ae8c57503c77e2cab6b5462ba",
  "testProject/newFile批把.txt": "2e65efe2a145dda7ee51d1741299f848e5bf752e",
  "testProject/package.json": "18a1e415e56220fa5122428a4ef8eb8874756576",
}
`;

exports[`getRepoStateAsync can parse committed files 1`] = `
Object {
  "nestedTestProject/package.json": "18a1e415e56220fa5122428a4ef8eb8874756576",
  "nestedTestProject/src/file 1.txt": "c7b2f707ac99ca522f965210a7b6b0b109863f34",
  "testProject/file  2.txt": "a385f754ec4fede884a4864d090064d9aeef8ccb",
  "testProject/file1.txt": "c7b2f707ac99ca522f965210a7b6b0b109863f34",
  "testProject/file蝴蝶.txt": "ae814af81e16cb2ae8c57503c77e2cab6b5462ba",
  "testProject/package.json": "18a1e415e56220fa5122428a4ef8eb8874756576",
}
`;

exports[`getRepoStateAsync handles requests for additional files 1`] = `
Object {
  "nestedTestProject/package.json": "18a1e415e56220fa5122428a4ef8eb8874756576",
  "nestedTestProject/src/file 1.txt": "c7b2f707ac99ca522f965210a7b6b0b109863f34",
  "testProject/file  2.txt": "a385f754ec4fede884a4864d090064d9aeef8ccb",
  "testProject/file1.txt": "c7b2f707ac99ca522f965210a7b6b0b109863f34",
  "testProject/file蝴蝶.txt": "ae814af81e16cb2ae8c57503c77e2cab6b5462ba",
  "testProject/log.log": "2e65efe2a145dda7ee51d1741299f848e5bf752e",
  "testProject/package.json": "18a1e415e56220fa5122428a4ef8eb8874756576",
}
`;

exports[`parseGitHashObject can parse multiple entries 1`] = `
Map {
  "a" => "11",
  "b" => "22",
  "c" => "33",
}
`;

exports[`parseGitHashObject can parse multiple entries with trailing whitespace 1`] = `
Map {
  "a" => "11",
  "b" => "22",
  "c" => "33",
}
`;

exports[`parseGitHashObject throws if too few hashes are provided 1`] = `"Expected 3 hashes from \\"git hash-object\\" but received 2"`;

exports[`parseGitHashObject throws if too many hashes are provided 1`] = `"Expected 2 hashes from \\"git hash-object\\" but received 3"`;
